apps/src/lib/node/ledger/mod.rs

```diff
@@ -350,9 +345,25 @@ async fn run_aux(config: config::Ledger, wasm_dir: PathBuf) {
             res
         });
 
-        let oracle = local.spawn_local(async move {
-            ethereum_node::run_oracle(&ethereum_url, eth_sender, abort_sender)
-                .await
+        let oracle = tokio::task::spawn_blocking(move || {
+            let rt = tokio::runtime::Handle::current();
+            rt.block_on(async move {
+                let local = tokio::task::LocalSet::new();
+                local
+                    .run_until(async move {
+                        tracing::info!("Ethereum event oracle is starting");
+                        ethereum_node::run_oracle(
+                            &ethereum_url,
+                            eth_sender,
+                            abort_sender,
+                        )
+                        .await;
+                        tracing::info!(
+                            "Ethereum event oracle is no longer running"
+                        );
+                    })
+                    .await
+            });
```

Comment:
we can probably move this `LocalSet` logic to `ethereum_node::run_oracle`. better yet, move the whole `tokio::task::spawn_blocking` block there, so this ugly hack is not exposed here.

Reply:
we would return a `JoinHandle` from `ethereum_node::run_oracle`

Reply:
yeah, it's necessary because of the `web30` crate.